vulnerabilities/importers/alpine_linux.py

```diff
@@ -20,124 +20,192 @@
-
-        if record["packages"] is None:
-            return advisories
-
-        for p in record["packages"]:
-            advisories.extend(
-                self._load_advisories(
-                    p["pkg"],
-                )
-            )
-
+            record = fetch_response(link).json()
+            if not record["packages"]:
+                raise Exception(f'"packages" not found in {link}')
+            advisories.extend(process_record(record))
         return advisories
 
-    @staticmethod
-    def _load_advisories(
-        pkg_infos: Mapping[str, Any],
-    ) -> List[Advisory]:
 
-        advisories = []
-
-        for fixed_vulns in pkg_infos["secfixes"].values():
-
-            if fixed_vulns is None:
-                continue
-
-            for vuln_ids in fixed_vulns:
-                vuln_ids = vuln_ids.split()
-                references = []
-                for reference_id in vuln_ids[1:]:
-
-                    if reference_id.startswith("XSA"):
-                        xsa_id = reference_id.split("-")[-1]
-                        references.append(
-                            Reference(
-                                reference_id=reference_id,
-                                url="https://xenbits.xen.org/xsa/advisory-{}.html".format(xsa_id),
-                            )
-                        )
-
-                    elif reference_id.startswith("ZBX"):
-                        references.append(
-                            Reference(
-                                reference_id=reference_id,
-                                url="https://support.zabbix.com/browse/{}".format(reference_id),
-                            )
+def fetch_response(url):
+    """
+    Fetch and return `response` from the `url`
+    """
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response
+    raise Exception("Failed to fetch data from the URL")
+
+
+def fetch_advisory_directory_links(page_response_content):
+    """
+    Return `advisory_directory_links` present in `index_page`
+    """
+    index_page = BeautifulSoup(page_response_content, features="lxml")
+    alpine_versions = [link.text for link in index_page.find_all("a") if link.text.startswith("v")]
+
+    assert alpine_versions, f"No versions found in {BASE_URL}"
+
+    advisory_directory_links = [f"{BASE_URL}{version}" for version in alpine_versions]
+
+    return advisory_directory_links
+
+
+def fetch_advisory_links(advisory_directory_page, advisory_directory_link):
+    """
+    Yield json file urls present in `advisory_directory_page`
+    """
+    advisory_directory_page = BeautifulSoup(advisory_directory_page, features="lxml")
+    anchor_tags = advisory_directory_page.find_all("a")
+    assert anchor_tags, f"No anchor tags found in {advisory_directory_link}"
+    for anchor_tag in anchor_tags:
+        if anchor_tag.text.endswith("json"):
+            yield f"{advisory_directory_link}{anchor_tag.text}"
```

Comment:
imo [urllib.parse.urljoin](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) or `requests.compat.urljoin` will be a safer way to concat urls